hostinfo/hostinfo_windows.go

```diff
@@ -21,19 +22,37 @@ func osVersionWindows() string {
 	if s, ok := winVerCache.Load().(string); ok {
 		return s
 	}
-	cmd := exec.Command("cmd", "/c", "ver")
-	cmd.SysProcAttr = &syscall.SysProcAttr{HideWindow: true}
-	out, _ := cmd.Output() // "\nMicrosoft Windows [Version 10.0.19041.388]\n\n"
-	s := strings.TrimSpace(string(out))
-	s = strings.TrimPrefix(s, "Microsoft Windows [")
-	s = strings.TrimSuffix(s, "]")
-
-	// "Version 10.x.y.z", with "Version" localized. Keep only stuff after the space.
-	if sp := strings.Index(s, " "); sp != -1 {
-		s = s[sp+1:]
+	major, minor, build := windows.RtlGetNtVersionNumbers()
+	s := fmt.Sprintf("%d.%d.%d", major, minor, build)
+	// Windows 11 still uses 10.0 as its (major.minor) internally
+	if major == 10 && minor == 0 {
```

Comment:
do you need to check minor == 0 or is major == 10 enough?

Reply:
Here's the dilemma that I've had in mind: as an undocumented registry value, this value isn't really formalized anywhere, so it is unclear whether it will work beyond the 10.0 major.minor pair. I suppose if it fails, it's just going to fail silently, so it doesn't matter that much. What do you think, @bradfitz ?

Reply:
Well, it's not like you're panicking the binary on failure. You're just conditionally appending it to the string, so I think it's fine to always do (well, when major == 10) and append it on err == nil.

Reply:
SGTM.